packages/react-menu/src/components/MenuTrigger/useMenuTrigger.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { MenuTriggerProps, MenuTriggerState } from './MenuTrigger.types';
+import { useMenuContext, MenuContextValue } from '../../menuContext';
+
+export const menuTriggerShorthandProps: (keyof MenuTriggerProps)[] = [];
+
+const mergeProps = makeMergeProps<MenuTriggerState>({ deepMerge: menuTriggerShorthandProps });
+
+/**
+ * Create the state required to render MenuTrigger.
+ * Clones the only child component and adds necessary event handling behaviours to open a popup menu
+ *
+ * @param props - props from this instance of MenuTrigger
+ * @param ref - reference to root HTMLElement of MenuTrigger
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory MenuTrigger }
+ */
+export const useMenuTrigger = (
+  props: MenuTriggerProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: MenuTriggerProps,
+): MenuTriggerState => {
+  const setOpen = useMenuContext(context => context.setOpen);
+
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
+    },
+    defaultProps,
+    resolveShorthandProps(props, menuTriggerShorthandProps),
+  );
+
+  state.setOpen = setOpen;
+
+  const child = React.Children.only(state.children);
+  state.children = React.cloneElement(child as React.ReactElement, getTriggerProps(state.setOpen));
```

Comment:
_tldr: feel free to ignore this PR_

---

A `React.Fragment` will be consider as a valid child, but we cannot assign props it:

```tsx
<MenuTrigger>
  <>
    <button /> // s_tldr: feel free to ignore this PR_

---

A `React.Fragment` will be consider as a valid child, but we cannot assign props it:

```tsx
<MenuTrigger>
  <> // 🚨 madness is there as we will clone a fragment
    <button />
    <button />
  </>
</MenuTrigger>
```

See https://github.com/Semantic-Org/Semantic-UI-React/pull/3998 as it tackles this issue. On N* side we also have a unified hook to handle triggers, see [`useTriggerElement.tsx`](https://github.com/microsoft/fluentui/blob/d5a8c91597cd66c0e1e03ac7cb4d8ada7e0d1b30/packages/fluentui/react-bindings/src/hooks/useTriggerElement.tsx)

Reply:
This will be handled in a follow up PR to set the trigger appropriately as I mentioned in another comment